crates/octez/src/async_node.rs

```diff
@@ -0,0 +1,73 @@
+use std::{fs::File, path::PathBuf, process::Stdio};
+
+use tokio::process::{Child, Command};
+
+use super::path_or_default;
+use anyhow::Result;
+
+pub struct AsyncOctezNode {
+    /// Path to the octez-node binary
+    /// If None, the binary will inside PATH will be used
+    pub octez_node_bin: Option<PathBuf>,
+    /// Path to the octez-node directory
+    pub octez_node_dir: PathBuf,
+}
+
+impl AsyncOctezNode {
+    fn command(&self) -> Command {
+        Command::new(path_or_default(self.octez_node_bin.as_ref(), "octez-node"))
+    }
+
+    pub async fn config_init(
+        &self,
+        network: &str,
+        rpc_endpoint: &str,
+        num_connections: u32,
+    ) -> Result<Child> {
+        Ok(self
+            .command()
+            .args([
+                "config",
+                "init",
+                "--network",
+                network,
+                "--data-dir",
+                self.octez_node_dir.to_str().expect("Invalid path"),
+                "--rpc-addr",
+                rpc_endpoint,
+                "--connections",
+                num_connections.to_string().as_str(),
+            ])
+            .spawn()?)
+    }
+
+    pub async fn generate_identity(&self) -> Result<Child> {
+        Ok(self
+            .command()
+            .args([
+                "identity",
+                "generate",
+                "0",
+                "--data-dir",
+                self.octez_node_dir.to_str().expect("Invalid path"),
+            ])
+            .spawn()?)
+    }
+
+    pub async fn run(&self, log_file: &File, options: &[&str]) -> Result<Child> {
+        let mut command = self.command();
+
+        command
+            .args([
+                "run",
+                "--data-dir",
+                self.octez_node_dir.to_str().expect("Invalid path"),
+                "--singleprocess",
+            ])
+            .args(options)
+            .stdout(Stdio::from(log_file.try_clone()?))
+            .stderr(Stdio::from(log_file.try_clone()?));
+
+        Ok(command.spawn()?)
```

Comment:
i think we should check the status of the process, unleess exit code is 0, throw 
```
    let status = command.spawn()?.wait().await?;
        match status.code() {
            Some(0) => Ok(()),
            _ => bail!("failed to generate keys for {}", alias),
        }
```

Reply:
I don't think this is needed here because `run` returns the only the child process. Whoever calls this can implement what you described. Also I think it's better to keep this behaviour the same as what is done in the synchronous node implementation now as eventually this async node will replace it and I don't want to add more logic to all references to accommodate this behaviour change when that takes place. We can do that afterwards if it's necessary.

Reply:
oh i meant to comment on the `generate identity` / `config init` as it could fail to init the config file if it already etc